### PR TITLE
HDDS-4811.Use OM style config to construct RaftGroup and initialize Raft Servers

### PR DESCRIPTION
**Pull request by amaliujia**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4811

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by amaliujia** [comment]
R: @mukul1987 @bshashikant

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java [hunk 1]
```diff
@@ -313,6 +313,13 @@ public RaftPeerId getPeerId() {
       List<RaftPeer> raftPeers = new ArrayList<>();
       // Add this Ratis server to the Ratis ring
       raftPeers.add(localRaftPeer);
+      for (SCMNodeDetails nodeDetails : details.getPeerNodeDetails()) {
```
Comment (bshashikant) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java:316]: I think , we need to add the peers depending upon whether SCM needs to start as single node server(1st regular start) or participate in 3 node ring(subsequent restarts). Let's hold on to this change for now and i will take care of this in the bootstrap sequence.

**Review by bshashikant — COMMENTED** [review]
@amaliujia , thanks for the patch. I see a TODO marked in SCMHANodeDetails#174 regarding details missing for Raft peer configuration.

Can you please specify what is missing here and address it in this patch?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java (hunk 1 above)
Comment (amaliujia) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java:316]: Sure sounds good.

**Comment by amaliujia** [comment]
@bshashikant 

Regrading  the SCMHANodeDetails#174 TODO, I used to think we load Raft config into SCMHANodeDetails directly. however, now the code is refactored as https://github.com/apache/ozone/blob/b60a2306374533a616123d1f9da33e0506bac4d6/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java#L93

So I guess there is no need for the SCMHANodeDetails#174 TODO.

At this moment HDDS-4811 maybe is already just a no-op JIRA

**Comment by bshashikant** [comment]
This change is not required any more as the required changes are done as part of https://github.com/apache/ozone/pull/1953.

Thanks @amaliujia for the efforts.

**Pull request closed without merging**